src/test/e2e/38_ephemeral_container_test.go

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package test provides e2e tests for Zarf.
+package test
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEphemeralContainers(t *testing.T) {
+	t.Log("E2E: Ephemeral Containers mutation")
+
+	tmpdir := t.TempDir()
+
+	// we need to create a test package that contains the images we want to potentially use
+	// this should ideally be a single pod such that naming is static
+	stdOut, stdErr, err := e2e.Zarf(t, "package", "create", "src/test/packages/38-ephemeral-container", "-o", tmpdir, "--skip-sbom")
+	require.NoError(t, err, stdOut, stdErr)
+	packageName := fmt.Sprintf("zarf-package-basic-pod-%s-0.0.1.tar.zst", e2e.Arch)
+	path := filepath.Join(tmpdir, packageName)
+
+	// deploy the above package
+	stdOut, stdErr, err = e2e.Zarf(t, "package", "deploy", path, "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+
+	// cleanup - should perform cleanup in the event of pass or fail
+	t.Cleanup(func() {
```

Comment:
This may be a matter of taste, but I'd rather we decouple cleanup from the order of the test and don't ensure that it didn't error. I'm assuming that this call will error out if it doesn't find `"basic-pod"` to delete, meaning we always have to call it after `package deploy` Making cleanup idempotent (doesn't matter how, where, or how many times we we call it, it just woks) makes the test more resilient to change.

I'm suggesting we should move this near the top of the function and declare all the expected resources we want to clean up after - errors can be ignored.